erpnext/selling/report/sales_details/sales_details.py

```diff
@@ -220,6 +220,9 @@ def get_conditions(self):
 		if self.filters.get("customer"):
 			conditions.append("s.customer = %(customer)s")
 
+		if self.filters.get("account_manager"):
+			conditions.append("cus.account_manager = %(account_manager)s")
```

Comment:
use tree condition, see for reference: https://github.com/ParaLogicTech/erpnext/blob/05111978ccefb748b891317bc03a9fb136096c58/erpnext/selling/report/sales_details/sales_details.py#L263-L266